system/src/iosMain/com.splendo.kaluga/system/network/NetworkManager.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ Copyright 2020 Splendo Consulting B.V. The Netherlands
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+
+package com.splendo.kaluga.system.network
+
+import co.touchlab.stately.concurrency.AtomicReference
+import com.splendo.kaluga.base.IOSVersion
+import com.splendo.kaluga.system.network.services.NetworkManagerHandler
+
+actual class NetworkManager actual constructor(
+    private val context: Any?,
+    private val onNetworkStateChange: NetworkStateChange
+) : BaseNetworkManager(onNetworkStateChange) {
+
+    private var _networkManagerHandler = AtomicReference<NetworkManagerHandler?>(null)
+    private var networkManagerHandler: NetworkManagerHandler?
+        get() = _networkManagerHandler.get()
+        set(value) = _networkManagerHandler.set(value)
+
+    override suspend fun startMonitoringNetwork() {
+        if (IOSVersion.systemVersion > IOSVersion(12)) {
```

Comment:
Just a thought: this class can be `open` and overriden for the right iOS version, and the Builder you still have to make (to get rid of context) will decide which version is used.

Probably doesn't need `expect/actual` anymore then.